app/channel.go

```diff
@@ -2578,7 +2578,7 @@ func (a *App) MarkChannelAsUnreadFromPost(postID string, userID string, collapse
 		}
 	}
 
-	channelUnread, nErr := a.Srv().Store.Channel().UpdateLastViewedAtPost(post, userID, unreadMentions, unreadMentionsRoot, *a.Config().ServiceSettings.ThreadAutoFollow, true)
```

Comment:
The buggy code that you removed was hiding the bug here.

 When `ThreadAutoFollow` is `true` and if the code you removed in `UpdateUnreadsByChannel` actually set the `LastViewed` instead of `LastUpdated`, then we'd see threads with replies newer than the post being marked as unread, being marked as unread. I checked the spreadsheet defining CRT behaviour and I dont think this was ever expected behaviour.